src/handlers/Ext01Handler.sol

```diff
@@ -143,8 +187,12 @@ contract Ext01Handler is OwnableUpgradeable, ReentrancyGuardUpgradeable, IExt01H
 
   function createExtOrder(
     CreateExtOrderParams memory _params
-  ) external payable nonReentrant returns (uint256 _orderId) {
+  ) external payable nonReentrant delegate(_params.mainAccount) returns (uint256 _orderIndex) {
     // Check
+    // Check if overrided _msgSender() is the same as _mainAccount.
+    // If msg.sender is not a delegatee, _msgSender() won't be overrided
+    // which then makes _msgSender() to become msg.sender not the _mainAccount.
+    if (_params.mainAccount != _msgSender()) revert IExt01Handler_Unauthorized();
     // 0 = Cancelled order
     if (_params.orderType == 0) revert IExt01Handler_BadOrderType();
     uint128 _minExecutionFee = minExecutionOrderOf[_params.orderType];
```

Comment:
Right now it is:

```solidity
if (msg.value != _minExecutionFee) revert IExt01Handler_InCorrectValueTransfer();
```

but I think it might be intended to be:

```solidity
if (msg.value != _params.executionFee) revert IExt01Handler_InCorrectValueTransfer();
```
Probably will be the same value most times, but do you think changing makes sense?

Reply:
Actually, I think it's important to change, because during a cancel the user would receive more ETH than they started with